### PR TITLE
Fixes for akka rewrite

### LINKED ISSUE #87: CrossCompat add compat import only if a symbol is replaced.

### PR DESCRIPTION
**Pull request by MasseGuillaume**
fix #87

With those changes, we can run the CrossCompat on Akka and compile the code without warning/errors and pass all the test.

Diff here:
https://github.com/MasseGuillaume/akka/commit/279fa620868fc80685a64951109bb2499af8f27b

### CODE REVIEW

File: scalafix/input/src/main/scala/fix/LinearSeqSrc.scala [hunk 1]
```diff
@@ -1,10 +0,0 @@
-/*
-rule = "scala:fix.CrossCompat"
- */
-package fix
-
-import scala.collection.LinearSeq
-
-object LinearSeqSrc {
-  LinearSeq(1)
-}
```
Comment (julienrf) [inline scalafix/input/src/main/scala/fix/LinearSeqSrc.scala]: We should still support this case

File: scalafix/rules/src/main/scala/fix/Stable212Base.scala [hunk 2]
```diff
@@ -45,17 +45,21 @@ trait Stable212Base extends CrossCompatibility { self: SemanticRule =>
   val traversable = exact(
     "_root_.scala.package.Traversable#",
     "_root_.scala.collection.Traversable#",
-    "_root_.scala.package.Iterable#",
-    "_root_.scala.collection.Iterable#"
```
Comment (julienrf) [inline scalafix/rules/src/main/scala/fix/Stable212Base.scala]: Why do we remove these?

File: scalafix/rules/src/main/scala/fix/Stable212Base.scala [hunk 3]
```diff
@@ -66,11 +70,6 @@ trait Stable212Base extends CrossCompatibility { self: SemanticRule =>
         "scala.collection.Traversable" -> "scala.collection.Iterable"
       )
 
-    val linearSeqToList =
-      ctx.replaceSymbols(
-        "scala.collection.LinearSeq" -> "scala.collection.immutable.List",
-      )
```
Comment (julienrf) [inline scalafix/rules/src/main/scala/fix/Stable212Base.scala]: Why remove this rule?

File: scalafix/input/src/main/scala/fix/LinearSeqSrc.scala (hunk 1 above)
Comment (MasseGuillaume) [reply scalafix/input/src/main/scala/fix/LinearSeqSrc.scala]: https://github.com/scala/scala-collection-compat/issues/99#issue-341891540

File: scalafix/rules/src/main/scala/fix/Stable212Base.scala (hunk 2 above)
Comment (MasseGuillaume) [reply scalafix/rules/src/main/scala/fix/Stable212Base.scala]: we use this to detect if we need to add the compat import for 

```scala
     ctx.replaceSymbols(
        "scala.Traversable"            -> "scala.Iterable",
        "scala.collection.Traversable" -> "scala.collection.Iterable"
      )
```

It was incorrectly adding compat import.

File: scalafix/rules/src/main/scala/fix/Stable212Base.scala (hunk 3 above)
Comment (MasseGuillaume) [reply scalafix/rules/src/main/scala/fix/Stable212Base.scala]: https://github.com/scala/scala-collection-compat/issues/99#issue-341891540

File: scalafix/input/src/main/scala/fix/LinearSeqSrc.scala (hunk 1 above)
Comment (MasseGuillaume) [reply scalafix/input/src/main/scala/fix/LinearSeqSrc.scala]: No, if it's incorrect we should remove it. I'm using it to migrate akka, if we keep this rewrite in It's not possible to use scala-collection-compat to rewrite code.

**Pull request closed without merging**